File: frontend/src/features/patient/hooks/GetPatient/GetPatientDto.ts
import type { Address, EmergencyContact } from "@/features/auth";
import type { MobilityStatus } from "@/features/patient/types/enums.types";
import type { LookupDto } from "../LookupDto";

export interface GetPatientDto {
  id: string;
  userId: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  dateOfBirth: string;
  gender: string;
  address: Address;
  emergencyContact: EmergencyContact;
  mobilityStatus?: MobilityStatus;
  allergies: LookupDto[];
  conditions: LookupDto[];
  medications: LookupDto[];
  bio?: string;
  profilePictureUrl?: string;
  distanceKm?: number | null;
}

File: backend/Modules/Patients/Modules.Patients.Features/GetPatient/GetPatientDto.cs
using Modules.Common.Features.ValueObjects;
using Modules.Patients.Domain.ValueObjects;

namespace Modules.Patients.Features.GetPatient;

public sealed record GetPatientDto(
    Guid Id,
    Guid UserId,
    string FirstName,
    string LastName,
    string Email,
    string PhoneNumber,
    string DateOfBirth,
    string Gender,
    Address Address,
    EmergencyContact EmergencyContact,
    MedicalInfo MedicalInfo,
    string? Bio,
    string? ProfilePictureUrl,
    double? DistanceKm = null);